src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2877,6 +2877,184 @@ export default class EthereumApi implements Api {
+   * @param newestBlock - Highest block of the requested range.
+   * @param rewardPercentiles - (Optional) A monotonically increasing list of percentile values. For each block in the requested range, the transactions will be sorted in ascending order by effective tip per gas and the corresponding effective tip for the percentile will be determined, accounting for gas consumed.
+   * @returns transaction base fee per gas and effective priority fee per gas for the requested/supported block range
+   *
+   * * `oldestBlock`:  - Lowest number block of the returned range.
+   * * `baseFeePerGas`:  - An array of block base fees per gas. This includes the next block after the newest of the returned range, because this value can be derived from the newest block. Zeroes are returned for pre-EIP-1559 blocks.
+   * * `gasUsedRatio`:  - An array of block gas used ratios. These are calculated as teh ratio of gasUsed and gasLimit.
+   * * `reward`:  - (optional) An array of effective priority fee per gas data points from a single block. All zeroes are returned if the block is empty.
+   *
+   * @example
+   * ```javascript
+   * const feeHistory = await provider.request({ method: "eth_feeHistory", params: ["0x5", "0xE5A51D", [10, 50, 80]] });
+   * console.log(feeHistory);
+   * ```
+   */
+  @assertArgLength(3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles: number[]
+  ): Promise<Ethereum.FeeHistory> {
+    const blockchain = this.#blockchain;
+    const PAD_PRECISION = 16;
+    const PRECISION_FLOAT = 1e14;
+    const PRECISION_BIG_INT = BigInt(1e16);
+
+    // if newestBlock is a tag, we already keep a ref in the Block Manager.
+    let newestBlockNumber;
+    const newBlock = await blockchain.blocks.getBlockByTag(newestBlock as Tag);
+
+    // If newestBlock is not a tag we can just use the newestBlock number.
+    if (newBlock) {
+      newestBlockNumber = newBlock.header.number.toNumber();
+    } else {
+      newestBlockNumber = Quantity.from(newestBlock).toNumber();
+    }
+
+    // blockCount > newestBlock is technically valid but we cannot go past the Genesis Block.
+    const totalBlocks = Math.min(
+      Quantity.toNumber(blockCount),
+      newestBlockNumber + 1
+    );
+
+    // Cut out early if no range is given.
+    if (totalBlocks === 0) {
+      return {
+        oldestBlock: Quantity.from(newestBlock).toString(),
+        baseFeePerGas: undefined,
+        gasUsedRatio: null,
+        reward: undefined
+      };
+    }
+
+    // blockCount is inclusive of newestBlock
+    const oldestBlockNumber = newestBlockNumber - (totalBlocks - 1);
+
+    const baseFeePerGas = new Array(totalBlocks);
+    const gasUsedRatio = new Array(totalBlocks);
+    let reward;
+    if (rewardPercentiles.length > 0) {
+      reward = new Array(totalBlocks);
+    }
+
+    let currentBlockNumber = oldestBlockNumber;
+    let currentBlock;
```

Comment:
```suggestion
    let currentBlock: Block;
```

This cascades down to the `maxPriorityFeePerGas` stuff where presently there's a bunch of implicit anys.